src/core/lv_indev.c

```diff
@@ -971,8 +971,8 @@ static void indev_proc_release(_lv_indev_proc_t * proc)
         lv_event_send(indev_obj_act, LV_EVENT_RELEASED, indev_act);
         if(indev_reset_check(proc)) return;
 
-        /*Send CLICK if no scrolling*/
-        if(scroll_obj == NULL) {
+        /*Send CLICK if no scrolling and no previous gesture detected*/
+        if(scroll_obj == NULL && proc->types.pointer.gesture_sent == 0) {
             if(proc->long_pr_sent == 0) {
```

Comment:
The condition should be added one line below to process the `else` part regardless to the gesture.
```suggestion
            if(proc->long_pr_sent == 0  && proc->types.pointer.gesture_sent == 0) {
```

Reply:
only this line is not enough? the click should be trig also ?